### PR TITLE
Features/gradex

### PR DESCRIPTION
**Pull request by root-daemon**
Implementing GradeX into AcademiaPro (RIP)
---
This pull request includes several changes to enhance the functionality and styling of the GradeX application. The most notable changes involve the addition of new components, updates to existing components, and configuration adjustments.

### New Components and Functionality:

* [`app/gradex/components/CourseCard.tsx`](diffhunk://#diff-e35d058ae258a28b2603373005baabbd2f9303f3553d0494bef7b5b7242bc282R1-R85): Added a new `MarkCard` component to display course details, current grade, and a slider for adjusting grades.
* [`app/gradex/page.tsx`](diffhunk://#diff-648cabbb5adeb2c02d5e12b16ffd939c05be1bca433492d1c04ba7831a0d419fR1-R104): Implemented the main GradeX page, including SGPA calculation and displaying `MarkCard` components for each course.

### UI Enhancements:

* [`components/ui/input.tsx`](diffhunk://#diff-c2f62fb0cb5e2955363d1c27d9cb0b33d03fed2193d16c3877af492d82770d2cR1-R22): Created a new `Input` component with custom styling for inputs.
* [`components/ui/slider.tsx`](diffhunk://#diff-07a292676501dd5df42ccea735847a828be18549b1d77c9d498b18a1b330e18cR1-R28): Added a new `Slider` component using `@radix-ui/react-slider` for enhanced slider functionality.
* [`app/globals.css`](diffhunk://#diff-79e0914d9fee5ca4432bdb002a24d700d31b5577265f4de3f90d5c292b9f1392R74-R76): Introduced a CSS variable for border radius.

### Configuration and Utilities:

* [`components.json`](diffhunk://#diff-2c9eb56f775dd960fa3fb60253b2d30f21baae2e81f352e63a0984f129408016R1-R21): Added a new configuration file for UI components with Tailwind and other settings.
* [`lib/utils.ts`](diffhunk://#diff-98e56006aa8453a29bc2b8bc8e6f718b3f17ba66470f3a52a5dc5bd643482e70R1-R6): Created a utility function `cn` for merging class names using `clsx` and `tailwind-merge`.
* [`tailwind.config.ts`](diffhunk://#diff-655dc9e3d0aa561e3fa164bf48bd89cb0f5da65e0a567f8ebbf9dd791a0e7f40L10-R10): Updated Tailwind configuration with custom settings and comments for clarity. [[1]](diffhunk://#diff-655dc9e3d0aa561e3fa164bf48bd89cb0f5da65e0a567f8ebbf9dd791a0e7f40L10-R10) [[2]](diffhunk://#diff-655dc9e3d0aa561e3fa164bf48bd89cb0f5da65e0a567f8ebbf9dd791a0e7f40L26-R26) [[3]](diffhunk://#diff-655dc9e3d0aa561e3fa164bf48bd89cb0f5da65e0a567f8ebbf9dd791a0e7f40L50-R60)

### Dependency Updates:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R13-R20): Added new dependencies including `@radix-ui/react-slider`, `clsx`, `tailwind-merge`, and `lucide-react`. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R13-R20) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L26-R32)

These changes collectively improve the user interface and functionality of the GradeX application, making it more interactive and visually appealing.

### CODE REVIEW

**Comment by root-daemon** [comment]
Code works UI is Ass, like so shit I wanna puke, maybe I already did, either way I will fix it later or more like never you know what you fix it, you will it change it anyway :)

**Comment by root-daemon** [comment]
Also we should possibly add a feature where we highlight the predict button or some shit so people will actually know we changed some shit.

File: tailwind.config.ts
Comment (Rahuletto) [inline tailwind.config.ts:1]: ???? Why commenting all of sudden

File: components/ui/input.tsx
Comment (Rahuletto) [inline components/ui/input.tsx:1]: Why do we need Input when we are just using the slider?

**Comment by Rahuletto** [comment]
I'll do the UI, leave it on me. Also Predict yea we can add a pseudo element :after and add "NEW"

**Pull request merged**